tasks/eva_2835/gather_per_species_per_assembly_variant_remapping_counts.py

```diff
@@ -0,0 +1,270 @@
+        cell_name = a_z[column + 2] + "1"
+        worksheet.write(cell_name, all_columns[column], header_format)
+
+    # Defining the cell format for the values
+    stats_fmt = workbook.add_format({'align': 'center', 'valign': 'vcenter', 'border': 1})
+    filename_fmt = workbook.add_format(
+        {'align': 'left', 'valign': 'vcenter', 'border': 1, 'font_color': 'red', 'text_wrap': 1})
+
+    # Setting the starting row for values
+    row = 1
+
+    # Populating the Excel with the values per taxonomy id and per assembly
+    for r in range(len(all_tax_id)):
+        worksheet.write(row, 0, all_tax_id[row - 1], filename_fmt)
+        worksheet.write(row, 1, all_assembly[row - 1], filename_fmt)
+
+        # Setting the index list which has values for a particular iteration
+        idx = []
+
+        for key in all_keys[row - 1]:
+            position = all_columns.index(key)
+            idx.append(position)
+
+        # Populating the values for the other stats columns for a particular iteration
+        for col in range(len(all_columns)):
+            if col in idx:
+                key = all_columns[col]
+                position = all_keys[row - 1].index(key)
+                stats = all_values[row - 1][position]
+                worksheet.write(row, col + 2, stats, stats_fmt)
+
+            else:
+                worksheet.write(row, col + 2, 0, stats_fmt)
+
+        # Incrementing the row to write the next record
+        row = row + 1
+
+    # Closing the workbook
+    workbook.close()
+
+
+def gather_counts_per_tax_per_assembly(path, taxid, assembly_accession):
+    """
+    This function is used to store the counts of the remapped variants along wih the reason for failures
+    at different rounds (i.e. for different lengths of the flanking region) for a particular taxonomy and
+    assembly
+
+    Input: The taxonomy id and the assembly accession from generate_output() along with the input files full
+           path
+
+    Output: All the statistics in global lists
+
+    """
+
+    # Setting the filename for the eva counts
+    filename_eva = str(assembly_accession) + "_eva_remapped_counts.yml"
+    filename_eva = os.path.join(path, str(taxid), str(assembly_accession), "eva", filename_eva)
+
+    # Setting the filename for the dbsnp counts
+    filename_dbsnp = str(assembly_accession) + "_dbsnp_remapped_counts.yml"
+    filename_dbsnp = os.path.join(path, str(taxid), str(assembly_accession), "dbsnp", filename_dbsnp)
+
+    # Calling functions for gathering stats per file from eva and dbsnp data
+    keys_values = gather_counts_per_file(filename_eva)
+    keys_values_dbsnp = gather_counts_per_file(filename_dbsnp)
+
+    # Adding the values of dbsnp and eva with the common keys
+    for key in keys_values_dbsnp:
+        if key in keys_values:
+            keys_values_dbsnp[key] = keys_values_dbsnp[key] + keys_values[key]
+
+    # Concatenating the two dictionaries
+    keys_values_dbsnp = {**keys_values, **keys_values_dbsnp}
```

Comment:
This works but I find it confusing and the name are not very descriptive.
I would suggest:
```suggestion
    all_headers = list(set(keys_values) + set(keys_values_dbsnp))
    all_values = {}
    for key in set(keys_values) + set(keys_values_dbsnp):
        if key in keys_values:
            all_values[key] = keys_values_dbsnp.get(key, 0) + keys_values.get(key, 0)
```

Reply:
I have now used more descriptive names

Reply:
I see.
My proposal was more about the way you create the merge in two loop where I do it in in one using default values when the key is not present.

Reply:
I have now incorporated the changes in the code for review